packages/experimental/Switch/src/Switch.tsx

```diff
@@ -14,9 +41,48 @@ export const Switch = compose<SwitchType>({
     toggleContainer: View,
     onOffText: Text,
   },
-  useRender: (_userProps: SwitchProps) => {
-    return (_final: SwitchProps) => {
-      return null; // Only implemented for Win32
+  useRender: (userProps: SwitchProps, useSlots: UseSlots<SwitchType>) => {
+    const animateSwitchKnob = () => {
+      if (Platform.OS === 'android') {
+        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
+      }
+    };
+
+    const switchInfo = useSwitch({
+      ...userProps,
+      onChange: (e: InteractionEvent, checked?: boolean) => {
+        animateSwitchKnob();
+        userProps.onChange && userProps.onChange(e, checked);
+      },
+    });
+
+    // grab the styled slots
+    const Slots = useSlots(userProps, (layer) => switchLookup(layer, switchInfo.state, switchInfo.props));
+
+    // Use Layout Animation for Android
+    if (Platform.OS === 'android' && UIManager.setLayoutAnimationEnabledExperimental) {
+      UIManager.setLayoutAnimationEnabledExperimental(true);
```

Comment:
Should this be turned off on dismount?

Reply:
Turned off once the animation completes.

Reply:
Changes done and approved.